assets/src/components/map.tsx

```diff
@@ -106,11 +107,33 @@ const Map = (props: Props): ReactElement<HTMLDivElement> => {
   const stops = (props.shapes || []).flatMap((shape) => shape.stops || [])
   const tileType = props.tileType || "base"
 
+  useEffect(() => {
+    const classList = mapRef.current?.getContainer().classList
+    if (props.extraClasses && classList) {
+      for (const [className, shouldBePresent] of Object.entries(
+        props.extraClasses
+      )) {
+        if (shouldBePresent && !classList.contains(className)) {
+          classList.add(className)
+        }
+        if (!shouldBePresent && classList.contains(className)) {
+          classList.remove(className)
+        }
+      }
+    }
+  }, [props.extraClasses, mapRef])
+
+  const classNames = props.extraClasses
+    ? Object.entries(props.extraClasses)
+        .filter(([_className, shouldBePresent]) => shouldBePresent)
+        .map(([className, _shouldBePresent]) => className)
+    : []
+
   return (
     <>
       <div className={stateClasses} />
       <MapContainer
-        className="c-vehicle-map"
+        className={joinClasses(["c-vehicle-map", ...classNames])}
```

Comment:
question: why are you using both `className` and `element.classList`?

Reply:
`className` here gives it the right classes when it first loads. `classList` lets it change those classes interactively in response to events.

User-facing effects:
- If I only use `className` and don't use `classList`, then when a user clicks to start a detour, the new fancy cursor doesn't show up.
- If I only use `classList` and don't use `className`, then it doesn't render correctly in storybook (and wouldn't render correctly if we ever started out with a detour in a partially-created state, say, after someone navigated back to an auto-saved draft)

Reply:
would you be open to/could you use `stateClasses` or wrap `<Map/>` in `<DetourMap/>` with a `div` which has these classes?

Reply:
Honestly yes - I don't love the solution I came up with here.

When I wrapped the `<Map />` in `<DetourMap />` in a div, the whole map disappeared, I think because it got a size of zero. That's easy to fix (I think), but when I saw that I got scared and ran the other way.

No idea what `stateClasses` are; happy to try them out though!

Reply:
> When I wrapped the `<Map />` in `<DetourMap />` in a div, the whole map disappeared, I think because it got a size of zero. That's easy to fix (I think), but when I saw that I got scared and ran the other way.

I _think_ if you added `h-100` to it, it'd not be size zero anymore


> No idea what `stateClasses` are; happy to try them out though!

It's a prop on `<Map/>` which adds classes on a `div` right next to the map. See https://github.com/mbta/skate/blob/main/assets/css/_leaflet.scss#L9-L10 for existing usages via sibling selectors

Reply:
I see I see!

[How does this look?](https://github.com/mbta/skate/pull/2466/commits/9510fdf56d6854c85df1a51a17ab31bc292ab241)

Reply:
LGTM!